packages/api-server/src/convert-tx.ts

```diff
@@ -367,11 +369,33 @@ export async function polyTxToGwTx(
     );
   }
 
+  // Check gas limit
   const gasLimitErr = verifyGasLimit(gasLimit === "0x" ? "0x0" : gasLimit, 0);
   if (gasLimitErr) {
     throw gasLimitErr.padContext(`eth_sendRawTransaction ${polyTxToGwTx.name}`);
   }
 
+  // only check if it is gasless transaction when entrypointContract is configured
+  if (
+    gwConfig.entrypointContract != null &&
+    isGaslessTransaction(
+      { to, gasPrice: gasPrice === "0x" ? "0x0" : gasPrice, data },
+      gwConfig.entrypointContract
+    )
+  ) {
+    const err = verifyGaslessTransaction(
+      to,
+      data,
+      gasPrice === "0x" ? "0x0" : gasPrice,
+      gasLimit === "0x" ? "0x0" : gasLimit,
+      0
+    );
+    if (err != null) {
+      throw err.padContext(`eth_sendRawTransaction ${polyTxToGwTx.name}`);
+    }
+  }
+
+  // Check gas price
```

Comment:
We should not check gas price again for `0 gas price` if it's a gasless tx.

Reply:
updated